packages/idempotency/tests/e2e/makeFunctionIdempotent.test.ts

```diff
@@ -0,0 +1,101 @@
+
+const app = new App();
+let stack: Stack;
+
+describe('Idempotency e2e test, default settings', () => {
+  const ddb = new DynamoDBClient({ region: 'eu-west-1' });
+  stack = new Stack(app, stackName);
+
+  const functionNameDefault = generateUniqueName(RESOURCE_NAME_PREFIX, v4(), runtime, 'default');
+  const ddbTableNameDefault = stackName + '-default-table';
+  createIdempotencyResources(stack, runtime, ddbTableNameDefault, makeFunctionIdepmpotentFile, functionNameDefault, 'handler');
+
+  const functionNameCustom = generateUniqueName(RESOURCE_NAME_PREFIX, v4(), runtime, 'custom');
+  const ddbTableNameCustom = stackName + '-custom-table';
+  createIdempotencyResources(stack, runtime, ddbTableNameCustom, makeFunctionIdepmpotentFile, functionNameCustom, 'handlerCustomized', 'customId');
+
+  const functionNameKeywordArg = generateUniqueName(RESOURCE_NAME_PREFIX, v4(), runtime, 'keywordarg');
+  const ddbTableNameKeywordArg = stackName + '-keywordarg-table';
+  createIdempotencyResources(stack, runtime, ddbTableNameKeywordArg, makeFunctionIdepmpotentFile, functionNameKeywordArg, 'handlerWithKeywordArgument');
+
+  beforeAll(async () => {
+
+    await deployStack(app, stack);
+
+  }, SETUP_TIMEOUT);
+
+  it('when called twice, it returns the same result', async () => {
+    const payload = { records: [ { id: 1, foo: 'bar' }, { id: 2, foo: 'baz' }, { id: 3, foo: 'bar' } ] };
+    await invokeFunction(functionNameDefault, 2, 'SEQUENTIAL', payload, false);
+
+    const payloadHashFirst = createHash('md5').update(JSON.stringify('bar')).digest('base64');
+    const payloadHashSecond = createHash('md5').update(JSON.stringify('baz')).digest('base64');
+
+    await ddb.send(new ScanCommand({ TableName: ddbTableNameDefault })).then((result) => {
+      expect(result?.Items?.length).toEqual(2);
+    });
+
+    await ddb.send(new GetCommand({
+      TableName: ddbTableNameDefault,
+      Key: { id: `${functionNameDefault}#${payloadHashFirst}` }
+    })).then((result) => {
+      expect(result?.Item?.data).toEqual('Processing done: bar');
+      expect(result?.Item?.status).toEqual('COMPLETED');
+
+    });
+
+    await ddb.send(new GetCommand({
+      TableName: ddbTableNameDefault,
+      Key: { id: `${functionNameDefault}#${payloadHashSecond}` }
+    })).then((result) => {
+      expect(result?.Item?.data).toEqual('Processing done: baz');
+      expect(result?.Item?.status).toEqual('COMPLETED');
+    });
+
+  }, TEST_CASE_TIMEOUT);
+
+  test('when called with customized function wrapper, it creates ddb entry with custom attributes', async () => {
+    const payload = { records: [ { id: 1, foo: 'bar' }, { id: 2, foo: 'baq' }, { id: 3, foo: 'bar' } ] };
+    const payloadHash = createHash('md5').update('"bar"').digest('base64');
+
+    const invocationLogsCustmozed = await invokeFunction(functionNameCustom, 2, 'SEQUENTIAL', payload, false);
+    await ddb.send(new GetCommand({
```

Comment:
You can't/shouldn't use `await` with `.then` afterwards.

Change this (& the other occurrences) to `const data = await ddb.send(...`